pkg/storer/internal/pinning/pinning.go

```diff
@@ -314,6 +318,20 @@ func DeletePin(ctx context.Context, st internal.Storage, root swarm.Address) err
 	return nil
 }
 
+func IterateCollectionChunks(st storage.Store, root swarm.Address, iterateFn func(swarm.Address) (bool, error)) error {
+	err := st.Iterate(storage.Query{
+		Factory:      func() storage.Item { return &pinChunkItem{UUID: root.Bytes()} },
```

Comment:
Not sure how this is working. The root is not the UUID. You need to read the UUID first.

Check export.go file in this pkg. It already has IterateCollection function. You can just move it here. Previously no one outside the pkg was using it. But now we do.